src/config/gtmMiddleware.js

```diff
@@ -0,0 +1,46 @@
+const { GA4_CONTAINER_ID } = require("./constants");
+const { routeInfo } = require("./constants");
+
+// This function makes sure that the GA4 Container ID is accessible to all pages, so we don't have to repeat it in every route
+
+const setGa4ContainerId = (req, res, next) => {
+  //     Set the GA4 Container ID in (locals) that all pages can see.
+  res.locals.ga4ContainerId = GA4_CONTAINER_ID;
+  next(); // Pass control to the next middleware function
+};
+
+// Middleware to instantiate the status code for the On Page Load tracker
+const setStatusCode = (req, res, next) => {
+  res.locals.statusCode = res.statusCode;
+  next();
+};
+
+// Middleware to instantiate the values for taxonomy levels 1 and 2 for the On Page Load tracker
+const setTaxonomyValues = (req, res, next) => {
+  const url = req.url;
+  const pathFound = routeInfo.find((route) => route.path === url);
+  if (pathFound) {
```

Comment:
You need an "else" for your "if".
If no path is found, what will be the behavior? With an "else", you can control it by affecting a default value (like "undefined") or/and adding a log with a specific message.
Same issue for pageTitle.

Reply:
I have added else statements for both if statements , setting the values to undefined if no path found and logging a message.